GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -405,11 +405,15 @@ public Process GetGitProcess(string command, string workingDirectory, string dot
 
             // Removing trace variables that might change git output and break parsing
             // List of environment variables: https://git-scm.com/book/gr/v2/Git-Internals-Environment-Variables
-            foreach (string key in processInfo.EnvironmentVariables.Keys.Cast<string>()
-                .Where(x => x.StartsWith("GIT_TRACE", StringComparison.OrdinalIgnoreCase))
-                .ToList())
+            foreach (string key in processInfo.EnvironmentVariables.Keys.Cast<string>())
             {
-                processInfo.EnvironmentVariables.Remove(key);
+                // If GIT_TRACE is set to a fully-rooted path, then Git sends the trace
+                // output to that path instead of stdout (GIT_TRACE=1) or stderr (GIT_TRACE=2).
+                if (key.StartsWith("GIT_TRACE", StringComparison.OrdinalIgnoreCase) &&
+                    !(key.Equals("GIT_TRACE") && Path.IsPathRooted(processInfo.EnvironmentVariables[key])))

```

Comment:
Any reason why we wouldn't want to leave it in fancy linq?
Aka
            foreach (string key in processInfo.EnvironmentVariables.Keys.Cast<string>()
                .Where(x => x.StartsWith("GIT_TRACE", StringComparison.OrdinalIgnoreCase) 
                    && !Path.IsPathRooted(processInfo.EnvironmentVariables[x]))
                .ToList())
            {
                processInfo.EnvironmentVariables.Remove(key);
            }

Also IsPathRooted can throw an exception if value contains invalid chars,  We should guard against that I think https://docs.microsoft.com/en-us/dotnet/api/system.io.path.ispathrooted?redirectedfrom=MSDN&view=netframework-4.7.2.

Reply:
Good find on that exception! I'll need to check for invalid chars first, I guess.

The extra complication we are doing here is something that is too large to fit nicely into a `.Where()` clause. It's also kind of pointless that we `.ToList()` something that is looped in a `foreach`.

Reply:
Thanks for going after this bug, it was actually a change I requested :-)

Why do you need the !(key.Equals(GIT_TRACE) bit?  

Agree we can drop the ToList.  It reads nicely to me in the Linq, but I'm happy to follow your lead here!

Reply:
There are other `GIT_TRACE_` variables, like `GIT_TRACE_PERF`, but I suppose we could stop interfering with those, too.